awx/api/serializers.py

```diff
@@ -98,21 +98,12 @@
                                            'total_hosts',
                                            'hosts_with_active_failures',
                                            'total_groups',
-                                           'groups_with_active_failures',
                                            'has_inventory_sources',
                                            'total_inventory_sources',
                                            'inventory_sources_with_failures',
                                            'organization_id',
                                            'kind',
                                            'insights_credential_id',),
-    'host': DEFAULT_SUMMARY_FIELDS + ('has_active_failures',
```

Comment:
`Host` loses all of these, and they're replaced w/ queries that generate the data from other sources:

https://github.com/ansible/awx/pull/5448/files#diff-a81324c523b41de7296fdd5ff9063d10R1748

Reply:
What's the performance differential when handling this in the list views?

Reply:
@matburt the prior case is going to be slightly faster on list views (it's just two boolean field lookups).

The after version has two *new* generated fields on the list view now:

```python
def get_has_active_failures(self, obj):
    return bool(
        obj.last_job_host_summary and obj.last_job_host_summary.failed
    )

def get_has_inventory_sources(self, obj):
     return obj.inventory_sources.exists()
```

Here's the `/api/v2/hosts/` list view *before* I preloaded my database with a *bunch* of data.  Note that I've set up an inventory w/ a bunch of hosts:

```sql
awx-dev=> SELECT COUNT(*) FROM main_host;
 count
-------
 22809
(1 row)
```

![image](https://user-images.githubusercontent.com/214912/71677318-8e85fa00-2d50-11ea-8848-737f81ae9d3c.png)

Reply:
Next, I made a bunch of inventory sources, and a ton of `JobHostSummary` objects associated with every host:

```python
In [1]: i = Inventory.objects.first()

In [2]: for s in InventorySource.objects.all():
   ...:     for h in Host.objects.all():
   ...:         h.inventory_sources.add(s)
   ...:
In [24]: for i in range(MANY):
    ...:     j = Job.objects.create()
    ...:     for h in Host.objects.all():
    ...:         j.job_host_summaries.get_or_create(host=h, host_name=h.name, defaults=host_stats)
    ...:
    ...:
```

```sql
awx-dev=> SELECT COUNT(*) FROM main_jobhostsummary;
 count
--------
 633365
(1 row)

awx-dev=> SELECT COUNT(*) FROM main_host_inventory_sources;
 count
-------
 132610
(1 row)
```

<img width="399" alt="image" src="https://user-images.githubusercontent.com/214912/71679461-a3b15780-2d55-11ea-8cac-b03028e9f812.png">

I'm going to keep running my data generator all day and see if I can get some of these tables into the millions and see if I notice any additional performance degradation.

Reply:
Update: I added some more prefetching to speed this up a bit, and it cut the query count in half:

https://github.com/ansible/awx/pull/5448/files#r362574316